atc/db/pipeline_factory.go

```diff
@@ -26,35 +26,21 @@ func NewPipelineFactory(conn Conn, lockFactory lock.LockFactory) PipelineFactory
 
 func (f *pipelineFactory) VisiblePipelines(teamNames []string) ([]Pipeline, error) {
 	rows, err := pipelinesQuery.
-		Where(sq.Eq{"t.name": teamNames}).
-		OrderBy("team_id ASC", "ordering ASC").
+		Where(sq.Or{
+			sq.Eq{"t.name": teamNames},
+			sq.And{
+				sq.NotEq{"t.name": teamNames},
+				sq.Eq{"public": true},
+			},
+		}).
+		OrderBy("public ASC", "team_id ASC", "ordering ASC").
```

Comment:
For the ordering, I'm not sure if this will result in the teams that you passed in to be ordered first, and then the public ones after because there can be times when the team you passed in is also public, and that will order it as part of the `public = true` portion.

You might be able to double check that the web UI relies on the API for the ordering or it might order the teams itself after it receives the response, that way we don't need to worry about the ordering.

Reply:
From the old code it seemed that we want the results to be ordered by public first, having the non-public ones at the beginning and then public ones after. I believe the unit tests and UI tests cover that case.

Reply:
The user experience this would affect is when viewing the sidebar as a non-super-admin; I can confirm that the dashboard does its own sorting. We didn't really lay out a requirement for the ordering in the sidebar though, so the UI definitely counts on the `/api/v1/pipelines` response to order pipelines. However #4483 and #4484 have just been opened, so there's definitely interest in defining this. It's your call if/when you want to make the simplification, but the sidebar for non-super-admins would definitely be affected if you drop that `public ASC` part.

Reply:
reverted the changes done here. So UI could expect the same ordering of pipelines still.